node/src/manager/commands/run.rs

```diff
@@ -31,6 +30,15 @@ use graph_core::{
 };
 use url::Url;
 
+fn locate(store: &dyn SubgraphStore, hash: &str) -> Result<DeploymentLocator, anyhow::Error> {
+    let mut locators = store.locators(&hash)?;
+    match locators.len() {
+        0 => bail!("could not find subgraph {hash} we just created"),
+        1 => Ok(locators.pop().unwrap()),
+        n => bail!("there are {n} subgraphs with hash {hash}"),
+    }
+}
+
```

Comment:
This _"get a single item from a collection"_ pattern happens so often that I created a convenience function for it [here](https://github.com/graphprotocol/graph-node/pull/3270/files#diff-e85bd4cf7aa1c61ad3464c284292e8adedecd56161573109222d0e397714ab6cR218-R230). 
In the future, I'll try to make that even more generic/flexible so we can reuse it.

Reply:
The reason this is special and not used with `DeploymentSearch` is that that wants the primary pool, and we only have a `SubgraphStore` here. The `run` code could probably be changed to also access the primary pool. But it's a smaller issue (and `run` needs a lot more TLC anyway)